MesaModel/model/SimModel.py

```diff
@@ -117,6 +119,33 @@ def __init__(
                 300,
             )
         )
+        self.ticks_per_home_day = self.model_params.ticks_per_home_day
+
+        if self.speedup > 1:
```

Comment:
Perhaps this new logic could live in its own function - the simulator's constructor strikes me as very long. Maybe a factory-like object would be useful for simulator one day, but I realise Mesa encourages this structure.

Reply:
Yes, the whole thing could probably do with a refactor but for now I've just moved the speedup calculation to its own function as you suggest.